### ISSUE TITLE
Unexpected behavior when skipping forward on Live TV stream

### ISSUE THREAD

**Issue by djp952** [djp952/pvr.hdhomerundvr#20]
From the SiliconDust forums:

> The only problem I see so far is with pausing Live TV for several minutes and then trying to go past
> the commercial with skipping forward. Something I use all the time.
> I have my remote setup so that I can skip forward by several minutes when I hit the skip function several times.
> However it only skips 10 seconds ahead even though I see it tries to go forward by showing "Seeking +03:00"
> It looks like it does not know exactly how big the buffer is at that point. Skipping forwards works great on recordings, but there is definitely a problem for me with Live TV buffer.

This is probably something that can be improved; the client is currently limiting how far into the future Kodi can request on a seek operation.  I believe with some tinkering I can improve this by letting Kodi ask for whatever it wants and let the RECORD engine handle it as best it can.  I will also test reporting an infinite length to Kodi for Live TV streams, IIRC that caused a problem at some point which was why I started tracking the "seen" length instead.

The code has changed/improved a lot since the last time I played with this, so I think it's reasonable to mark this as a bug/defect and work on it from that perspective.

**Comment by djp952** [comment]
I was able to correct this behavior as well as the poor behavior of in-progress recorded TV streams for version 1.2.6.  Will leave this out here until I get some feedback and close it if the feedback is positive.

**Comment by djp952** [comment]
Marking this as closed.  In general the functionality worked OK but needed a tweak to better handle a transition that occurs when an in-progress recording finishes on the back end.  HDHomeRun RECORD stops sending a "live" stream and sends a "recorded" stream which changes the Content-Range headers.

**Comment by hunsra** [comment]
Not sure if this is the same issue as reported above, but on all of my devices (gen 2 sticks, gen 2 fire tv boxes, and nvidia shield tv) fast forwarding a recorded stream beyond 2x seems to get kodi confused and causes it to take 10 or more seconds to recover from fast forwarding when resuming normal playback.  Audio starts right away, but video stays frozen on the last updated frame for 10 seconds or so, then syncs up with the audio and everything seems normal.

I have attached a debug log I captured during a test of this.  Not sure how I can help, but I'm willing to do anything to assist in troubleshooting.  This is the last thing I need to be able to confidently replace cable boxes with HD HomeRun Primes, Kodi, and the PVR/DVR plugin.

[kodi.zip](https://github.com/djp952/pvr.hdhomerundvr/files/1633465/kodi.zip)
